### PR TITLE
[Fix] pyOpenMS Plotting Documentation Errors & Warnings

### PR DESCRIPTION
**Pull request by matteopilz**
## Description
Removed the section underlines for all docstrings and changed the indentation. Should work now as intended.

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Review by jpfeuffer — CHANGES_REQUESTED** [review]
No, sorry, I do not think it will:
https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html

**Comment by matteopilz** [comment]
image

It should look like this right? That's from the API documentation page I have built with `make html`.

**Comment by jpfeuffer** [comment]
No, yours is regular RST that happens to look okish. (Your types are also not clickable).
Correctly parsed arguments look like this: https://pyopenms.readthedocs.io/en/latest/_autosummary/pyopenms.MzMLFile.html#pyopenms.MzMLFile.loadBuffer

File: src/pyOpenMS/pyopenms/plotting.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
-"""Plot MSSpectrum & MSChromatogram data
+"""
```
Comment (jpfeuffer) [inline src/pyOpenMS/pyopenms/plotting.py:1]: This is wrong. Summary belong in the first line.

File: src/pyOpenMS/pyopenms/plotting.py [hunk 2]
```diff
@@ -223,30 +223,32 @@ def _annotate_ion(mz: float, intensity: float, annotation: Optional[str],
     """
     Annotate a specific fragment peak.
```
Comment (jpfeuffer) [inline src/pyOpenMS/pyopenms/plotting.py:224]: Wrong.

File: src/pyOpenMS/pyopenms/plotting.py [hunk 3]
```diff
@@ -275,41 +277,42 @@ def _annotate_ion(mz: float, intensity: float, annotation: Optional[str],
 
 
 def plot_spectrum(spectrum: "MSSpectrum", color_ions: bool = True,
-             annotate_ions: bool = True, matched_peaks: Optional[Set] = None, annot_kws: Optional[Dict] = None,
-             mirror_intensity: bool = False, grid: Union[bool, str] = True, ax=None):
+                  annotate_ions: bool = True, matched_peaks: Optional[Set] = None, annot_kws: Optional[Dict] = None,
+                  mirror_intensity: bool = False, grid: Union[bool, str] = True, ax=None):
     """
     Plot an MS/MS spectrum.
```
Comment (jpfeuffer) [inline src/pyOpenMS/pyopenms/plotting.py:283]: Wrong

Comment (jpfeuffer) [reply src/pyOpenMS/pyopenms/plotting.py:283]: adapt to following

**Review by jpfeuffer — CHANGES_REQUESTED** [review]

File: src/pyOpenMS/pyopenms/plotting.py [hunk 4]
```diff
@@ -220,33 +218,34 @@ def plot_chromatogram(c: "MSChromatogram"):
...
+    """Annotate a specific fragment peak.
+
+    :param mz: The peak's m/z value (position of the annotation on the x axis).
+    :type mz: float
+
+    :param intensity: The peak's intensity (position of the annotation on the y axis).
+    :type intensity: float
+
+    :param annotation: The annotation that will be plotted.
+    :type annotation: str, optional
```
Comment (jpfeuffer) [inline src/pyOpenMS/pyopenms/plotting.py:230]: The sphinx docs are not very clear about this optional, but it looks correct. We need to check how this renders.

File: src/pyOpenMS/pyopenms/plotting.py [hunk 5]
```diff
@@ -275,41 +274,41 @@ def _annotate_ion(mz: float, intensity: float, annotation: Optional[str],
...
+    :type spectrum: MSSpectrum
+
+    :param color_ions: Flag indicating whether to color annotated fragment ions. The default is True.
+    :type color_ions: bool, optional
+
+    :param annotate_ions: Flag indicating whether to annotate fragment ions. The default is True.
+    :type annotate_ions: bool, optional
+
+    :param matched_peaks: Indices of matched peaks in a spectrum alignment.
+    :type matched_peaks: Optional[Set], optional
```
Comment (jpfeuffer) [inline src/pyOpenMS/pyopenms/plotting.py:292]: question is: is this one optional too much?

**Comment by jpfeuffer** [comment]
I will merge, since it definitely improves docs but we need to keep an eye on how this "optional" renders.

**Review by jpfeuffer — APPROVED** [review]

**Pull request merged**